SystemManager/Src/SystemManager.cpp

```diff
@@ -10,37 +10,56 @@
 #include "drivers_config.hpp"
 #include "independent_watchdog.h"
 #include "tim.h"
+#include "CommonDataTypes.hpp"
+#include "config.hpp"
+#include "cmsis_os.h"
 
 SystemManager::SystemManager():
     rcController_(&huart2),
     throttleMotorChannel_(&htim2, TIM_CHANNEL_1),
     yawMotorChannel_(&htim2, TIM_CHANNEL_2),
     rollMotorChannel_(&htim2, TIM_CHANNEL_3),
     pitchMotorChannel_(&htim2, TIM_CHANNEL_4),
-    invertedRollMotorChannel_(&htim3, TIM_CHANNEL_1),
-    watchdog_(&hiwdg)
-{}
+    watchdog_(&hiwdg),
+    am_instance_(config::flightmodes[config::DEFAULT_FLIGHTMODE].flightmodeConstructor())
+{
+    xTaskCreate(runAM, "AM Thread", 400U, (void*)&am_instance_, osPriorityNormal, &AM_handle_);
```

Comment:
I can't remember if anything else needs to be done to start a task (I don't think so though...). Were you able to test this on hardware?

Reply:
![image](https://github.com/UWARG/ZeroPilot-SW-3.5/assets/70457170/b52050d9-e50a-4b8a-97dc-5e570c68e2dd)

This looks promising. 

I'm running the AM thread at a period of 1 second and the set AM instructions thread at a period of 2 seconds (which simply increments the AM control struct using the static getter and setter methods defined in AM).

The AM thread `runControlLoopIteration()` seems to be picking up the new values at the regular interval.